vars/checksPublishResults.groovy

```diff
@@ -0,0 +1,151 @@
+        options.put('high', settings.get('high'))
+        options.put('normal', settings.get('normal'))
+        options.put('low', settings.get('low'))
+        // publish
+        step(options)
+    }
+}
+
+def report(publisherName, settings, doArchive){
+    if (settings.active) {
+        def pattern = settings.get('pattern')
+        def options = createCommonOptionsMap(publisherName, settings)
+        options.put('pattern', pattern)
+        // publish
+        step(options)
+        // archive check results
+        archiveResults(doArchive && settings.get('archive'), pattern, true)
+    }
+}
+
+def reportWarnings(parserName, settings, doArchive){
+    if (settings.active) {
+        def pattern = settings.get('pattern')
+        def options = createCommonOptionsMap('WarningsPublisher', settings)
+        options.put('parserConfigurations', [[
+            parserName: parserName,
+            pattern: pattern
+        ]])
+        // publish
+        step(options)
+        // archive check results
+        archiveResults(doArchive && settings.get('archive'), pattern, true)
+    }
+}
+
+def archiveResults(archive, pattern, allowEmpty){
+    if(archive){
+        echo "[${STEP_NAME}] archive ${pattern}"
+        archiveArtifacts artifacts: pattern, allowEmptyArchive: allowEmpty
+    }
+}
+
+@NonCPS
+def createCommonOptionsMap(publisherName, settings){
+    Map result = [:]
+    def thresholds = settings.get('thresholds', [:])
+    def fail = thresholds.get('fail', [:])
+    def unstable = thresholds.get('unstable', [:])
+
+    result.put('$class', publisherName)
+    result.put('healthy', settings.get('healthy'))
+    result.put('unHealthy', settings.get('unHealthy'))
+    result.put('canRunOnFailed', true)
+    result.put('failedTotalAll', fail.get('all'))
+    result.put('failedTotalHigh', fail.get('high'))
+    result.put('failedTotalNormal', fail.get('normal'))
+    result.put('failedTotalLow', fail.get('low'))
+    result.put('unstableTotalAll', unstable.get('all'))
+    result.put('unstableTotalHigh', unstable.get('high'))
+    result.put('unstableTotalNormal', unstable.get('normal'))
+    result.put('unstableTotalLow', unstable.get('low'))
+    // filter empty values
+    result = result.findAll {
+        return it.value != null && it.value != ''
+    }
+    return result
+}
+
+@NonCPS
+def prepare(parameters){
+    // ensure tool maps are initialized correctly
+    for(String tool : ['aggregation','tasks','pmd','cpd','findbugs','checkstyle','eslint','pylint']){
+        parameters[tool] = toMap(parameters[tool])
+    }
+    return parameters
+}
+
+@NonCPS
+def toMap(parameter){
+    if(MapUtils.isMap(parameter))
+        parameter.put('active', parameter.active == null?true:parameter.active)
+    else if(Boolean.TRUE.equals(parameter))
+        parameter = [active: true]
+    else if(Boolean.FALSE.equals(parameter))
+        parameter = [active: false]
+    else
+        parameter = [:]
```

Comment:
It is not misconfigured but simply not mentioned. 

So you can use `checksPublishResults(pmd: true)` and `checksPublishResults(pmd: false)` or `checksPublishResults(pmd: [pattern: 'foo'])` but also `checksPublishResults(checkstyle: true)` is valid. In this case we want to merge an empty config for `pmd` instead of failing. In this case the defaults will jump in.